### PR TITLE
Digitalpack mma frontend

### PR DESCRIPTION
**Pull request by tomverran**
This is the frontend for the digital pack MMA page
It attempts to minimise the amount of duplication between it and the membership page, hence renaming some styles from mem to mma and the expansion of the stripe form JS to also handle displaying the card element

image

### CODE REVIEW

**Comment by tomverran** [comment]
closed until I fix the karma tests

**Comment by tomverran** [comment]
cc @joelochlann

File: static/src/javascripts/bootstraps/digitalpack.js [hunk 1]
```diff
@@ -0,0 +1,3 @@
+define(['mmaTabs/digitalpack-tab'], function (digipackTab) {
```
Comment (joelochlann) [inline static/src/javascripts/bootstraps/digitalpack.js]: Why are these necessary? Can't we just require `mmaTabs/digitalpack-tab` directly in `account-profile.js`?

File: static/src/javascripts/projects/mma-tabs/digitalpack-tab.js [hunk 2]
```diff
@@ -0,0 +1,107 @@
...
+        DIGITALPACK_PRODUCT = '.js-dig-product',
+        UP_SELL = '.js-dig-up-sell',
+        DIG_INFO = '.js-dig-info',
+        LOADER = '.js-dig-loader',
+        IS_HIDDEN_CLASSNAME = 'is-hidden',
+        stripeForm = new PaymentForm();
+
+    function fetchUserDetails() {
+        ajax({
+
```
Comment (joelochlann) [inline static/src/javascripts/projects/mma-tabs/digitalpack-tab.js]: whitespace

**Comment by joelochlann** [comment]
I'm not too keen on the abbreviation `mma` everywhere - I know `manage-my-account` or `manageMyAccount` is a bit wordy but I feel that `mma` is cryptic enough to be frustrating when browsing through the codebase. Trying to see if I can think of labels more concise but as descriptive as `manage-my-account`

File: static/src/javascripts/projects/mma-tabs/digitalpack-tab.js [hunk 3]
```diff
@@ -0,0 +1,107 @@
...
+        DIGITALPACK_DETAILS = '.js-dig-details',
+        DIGITALPACK_PRODUCT = '.js-dig-product',
+        UP_SELL = '.js-dig-up-sell',
+        DIG_INFO = '.js-dig-info',
+        LOADER = '.js-dig-loader',
+        IS_HIDDEN_CLASSNAME = 'is-hidden',
+        stripeForm = new PaymentForm();
+
+    function fetchUserDetails() {
+        ajax({
```
Comment (joelochlann) [inline static/src/javascripts/projects/mma-tabs/digitalpack-tab.js]: There's actually a note in `ajax.js` about it being deprecating and preferring `ajax-promise.js` instead. Might be a good opportunity to switch. https://github.com/guardian/frontend/blob/master/static/src/javascripts/projects/common/utils/ajax.js#L12

File: static/src/javascripts/projects/mma-tabs/digitalpack-tab.js [hunk 4]
```diff
@@ -0,0 +1,107 @@
...
+    /**
+     * @param {{
+     *  optIn,
+     *  joinDate,subscriberId,
+     *  subscription: {trialLength, nextPaymentDate,nextPaymentPrice,renewalDate,
+     *  plan: {name,amount,interval}}
+     * }} userDetails
+     */
+    function populateUserDetails(userDetails) {
+        $(SUBSCRIBER_ID).text(userDetails.subscription.subscriberId);
```
Comment (joelochlann) [inline static/src/javascripts/projects/mma-tabs/digitalpack-tab.js]: As discussed, it would be much nicer to use a templating engine or some library with DOM data-binding or similar to handle all of this. I noticed lodash is already included in the project - this has a [basic templating engine](https://lodash.com/docs#template). Maybe it would be worth giving it a go?

File: static/src/javascripts/projects/mma-tabs/payment-form.js [hunk 5]
```diff
@@ -449,25 +494,44 @@ define([
...
     };
 
-    StripePaymentForm.prototype.init = function (form, successCallback) {
-        this.context = form;
-        this.successCallback = successCallback;
+    /**
+     * To display visa logo etc
+     */
+    StripePaymentForm.prototype.addIconCss = function () {
+        var spriteSheetUrl = $(this.context).data('sprite-url'),
```
Comment (joelochlann) [inline static/src/javascripts/projects/mma-tabs/payment-form.js]: Why do we need to dynamically pull in this stylesheet?

File: static/src/javascripts/projects/mma-tabs/payment-form.js [hunk 6]
```diff
@@ -449,25 +494,44 @@ define([
...
+        $(this.config.DOM.UPDATING).css('display', 'none');
     };
 
     StripePaymentForm.prototype.reset = function () {
-        this.getElement('CREDIT_CARD_IMAGE')[0].className = this.getElement('CREDIT_CARD_IMAGE')[0].className.replace(/\bi-\S*/gi, '');
-        this.getElement('CREDIT_CARD_NUMBER').val('');
-        this.getElement('CREDIT_CARD_CVC').val('');
-        this.getElement('CREDIT_CARD_EXPIRY_MONTH')[0].selectedIndex = 0;
-        this.getElement('CREDIT_CARD_EXPIRY_YEAR')[0].selectedIndex = 0;
+        $(this.config.DOM.CREDIT_CARD_IMAGE)[0].className = $(this.config.DOM.CREDIT_CARD_IMAGE)[0].className.replace(/\bi-\S*/gi, '');
```
Comment (joelochlann) [inline static/src/javascripts/projects/mma-tabs/payment-form.js]: Can we cache all these bonzo wrapped sets in the `.init()` function?

**Comment by joelochlann** [comment]
:+1: Looks OK and in keeping with surrounding style. Just a few questions & comments. Would definitely be worth getting the opinion of someone from frontend or identity, especially as regards where to draw the line with manual DOM manipulation versus using some better abstractions for highly dynamic parts of the site

**Comment by tomverran** [comment]
since this PR is really big I'm going to split it up and deploy it in several stages
